dali/kernels/erase/erase_cpu.h

```diff
@@ -0,0 +1,160 @@
+void EraseKernelImpl(T *data,
+                     const TensorShape<Dims> &strides,
+                     const TensorShape<Dims> &shape,
+                     const T* fill_values,
+                     int channels_dim,
+                     std::integral_constant<int, DimsLeft>) {
+  constexpr auto d = Dims - DimsLeft;  // NOLINT
+  for (int i = 0; i < shape[d]; i++) {
+    EraseKernelImpl(data, strides, shape, fill_values, channels_dim,
+                    std::integral_constant<int, DimsLeft - 1>());
+    data += strides[d];
+    if (d == channels_dim) {
+      fill_values++;
+    }
+  }
+}
+
+}  // namespace detail
+
+
+template <typename T, int Dims>
+void EraseKernel(T *data,
+                 const TensorShape<Dims> &strides,
+                 const TensorShape<Dims> &anchor,
+                 const TensorShape<Dims> &shape,
+                 const T* fill_values = nullptr,
+                 int channels_dim = -1) {
+  T default_fill_value = 0;
+  if (fill_values == nullptr) {
+    fill_values = &default_fill_value;
+    assert(channels_dim == -1);
+  }
+  for (int d = 0; d < Dims; d++) {
+    data += strides[d] * anchor[d];
+  }
+  detail::EraseKernelImpl(data, strides, shape, fill_values, channels_dim,
+                          std::integral_constant<int, Dims>());
+}
+
+template <typename T, int Dims>
+class EraseCpu {
+ public:
+  KernelRequirements Setup(KernelContext &context,
+                           const InTensorCPU<T, Dims> &in,
+                           const EraseArgs<T, Dims> &args) {
+    KernelRequirements req;
+    req.output_shapes.push_back(uniform_list_shape<Dims>(1, in.shape));
+    return req;
+  }
+
+  void Run(KernelContext &context,
+           OutTensorCPU<T, Dims> &out,
+           const InTensorCPU<T, Dims> &in,
+           const EraseArgs<T, Dims> &orig_args) {
+    auto args = orig_args;
+    DALI_ENFORCE(in.shape == out.shape);
+    const auto &shape = out.shape;
+    auto strides = GetStrides(shape);
+    const T *in_ptr = in.data;
+    T *out_ptr = out.data;
+    if (out_ptr != in_ptr) {
+      std::memcpy(out_ptr, in_ptr, volume(shape) * sizeof(T));
+    }
+
+    for (auto &roi : args.rois) {
+      bool valid_region = true;
+      for (int d = 0; d < Dims; d++) {
+        // correcting anchor if out of bounds
+        if (roi.anchor[d] < 0) {
+          roi.shape[d] += roi.anchor[d];
+          roi.anchor[d] = 0;
+        }
+        // correcting shape if out of bounds
+        if ((roi.anchor[d] + roi.shape[d]) > shape[d]) {
+          roi.shape[d] = shape[d] - roi.anchor[d];
+          assert(roi.anchor[d] + roi.shape[d] == shape[d]);
+        }
+
+        // filter out invalid regions
+        if (roi.shape[d] < 1) {
+          valid_region = false;
+        }
+
+        // at this point the region should be within bounds
+        assert(roi.anchor[d] >= 0);
+        assert(roi.anchor[d] + roi.shape[d] <= shape[d]);
+      }
+
+      if (!valid_region)
+        continue;
```

Comment:
Do we want to drop or assert?

Reply:
I'd say that for now ignoring the invalid regions should be enough. If we want more validation we can always extend this later

Reply:
ok